plugins/modules/consul_auth_method.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2024, Florian Apolloner (@apollo13)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: consul_auth_method
+short_description: Manipulate Consul auth methods
+version_added: 8.3.0
+description:
+ - Allows the addition, modification and deletion of auth methods in a consul
+   cluster via the agent. For more details on using and configuring ACLs,
+   see U(https://www.consul.io/docs/guides/acl.html).
+author:
+  - Florian Apolloner (@apollo13)
+extends_documentation_fragment:
+  - community.general.consul
+  - community.general.consul.token
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: partial
+    details:
+      - In check mode the diff will miss operational attributes.
+options:
+  state:
+    description:
+      - Whether the token should be present or absent.
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  name:
+    description:
+      - Specifies a name for the ACL auth method.
+      - The name can contain alphanumeric characters, dashes C(-), and underscores C(_).
+    type: str
+    required: true
+  type:
+    description:
+      - The type of auth method being configured.
+      - This field is immutable.
+      - Required when the auth method is created.
+    type: str
+    choices: ['kubernetes', 'jwt', 'oidc', 'aws-iam']
+  description:
+    description:
+      - Free form human readable description of the auth method.
+    type: str
+  display_name:
+    description:
+      - An optional name to use instead of O(name) when displaying information about this auth method.
+    type: str
+  max_token_ttl:
+    description:
+      - This specifies the maximum life of any token created by this auth method.
+      - Can be specified in the form of V(60s) or V(5m) (that is, 60 seconds or 5 minutes, respectively).
+    type: str
+  token_locality:
+    description:
+      - Defines the kind of token that this auth method should produce.
+    type: str
+    choices: ['local', 'global']
+  config:
+    description:
+      - The raw configuration to use for the chosen auth method.
+      - Contents will vary depending upon the type chosen.
+      - Required when the auth method is created.
+    type: dict
+"""
+
+EXAMPLES = """
+- name: create an auth method
```

Comment:
Task names should always start with an uppercase letter:
```suggestion
- name: Create an auth method
```